packages/lib/getMessagesForMaxToken.ts

```diff
@@ -0,0 +1,27 @@
+import { HumanMessage } from 'langchain/schema';
+
+import countTokens from './count-tokens';
+
+export default function getMessagesForMaxToken({
+  messages,
+  maxTokens,
+}: {
+  messages: HumanMessage[];
+  maxTokens: number;
+}) {
+  const messagesWithinTokenLimit = [];
+  let currentTokenCount = 0;
+
+  for (const message of messages) {
+    const messageTokens = countTokens({ text: message.content });
+    currentTokenCount += messageTokens;
+
+    if (currentTokenCount <= maxTokens) {
+      messagesWithinTokenLimit.push(message);
+    } else {
```

Comment:
Would be great to go one step further and truncate a message to fill maxTokens
example
maxTokens = 1000
messages -> length 1, message.content is 2000 tokens

The function should return 1 message 1000 tokens length

🤖